test/cpp/end2end/xds/BUILD

```diff
@@ -156,6 +156,33 @@ grpc_cc_test(
     ],
 )
 
+grpc_cc_test(
+    name = "xds_cluster_override_host_end2end_test",
```

Comment:
Instead of creating a new file, let's just add these tests to the existing xds_override_host_end2end_test.cc file.  The functionality we're adding here is really part of that same feature, so it makes sense to have all of the tests in one place.